src/wp-includes/blocks.php

```diff
@@ -1002,6 +1002,120 @@ function set_ignored_hooked_blocks_metadata( &$parsed_anchor_block, $relative_po
 	return '';
 }
 
+/**
+ * Runs the hooked blocks algorithm on the given content.
+ *
+ * This function is meant for internal use only.
+ *
+ * @since 6.6.0
+ * @access private
+ *
+ * @param string $content Serialized content.
+ * @param WP_Block_Template|WP_Post|array $context       A block template, template part, `wp_navigation` post object,
+ *                                                       or pattern that the blocks belong to.
+ * @return string The serialized markup with the ignored hooked blocks metadata applied.
+ */
+function insert_hooked_blocks_into_content( $content, $context, $callback = 'insert_hooked_blocks' ) {
```

Comment:
I need to retract my earlier naming suggesting 😭 I realized that the function's behavior is dependent on the `$callback`, so it doesn't always insert hooked blocks.

I guess "applying" something "recursively" to "content", or to a "block tree", are better keywords here. We can revert back to `apply_hooked_blocks_to_content`. Or maybe `apply_block_hooks_to_content`? (As "Block Hooks" is the algorithm we're applying; whereas hooked blocks we aren't always adding/applying) 🤔

Reply:
Yeah, good catch as that had totally bypassed me too. Updated in https://github.com/WordPress/wordpress-develop/pull/6604/commits/08e459bb237c8540b6e3c337c9ddb4c92587bdb6